library/std/src/sys/hermit/mod.rs

```diff
@@ -26,11 +26,13 @@ pub mod env;
 pub mod ext;
 pub mod fd;
 pub mod fs;
+#[path = "../unsupported/io.rs"]
 pub mod io;
 pub mod memchr;
 pub mod mutex;
 pub mod net;
 pub mod os;
+#[path = "../unix/path.rs"]
```

Comment:
Hm, I'm a bit worried that in theory people changing the unix path.rs will not expect it to also get picked up on other platforms. It does seem pretty unlikely to get changed, but maybe we can at least add a comment to that file that it's also used by other platforms?